lib/scaffold/start.js

```diff
@@ -36,18 +36,23 @@ function start(options) {
           bitcoreNodeModule = moduleName + '/' + modulePackage.bitcoreNode;
         }
         bitcoreModule = require(bitcoreNodeModule);
-
       }
 
       // check that the module supports expected methods
       if (!bitcoreModule.prototype ||
+          !bitcoreModule.dependencies ||
           !bitcoreModule.prototype.start ||
           !bitcoreModule.prototype.stop) {
         throw new Error(
           'Could not load module "' + moduleName + '" as it does not support necessary methods.'
         );
       }
-      bitcoreModules.push(bitcoreModule);
+      bitcoreModules.push({
+        name: moduleName,
+        module: bitcoreModule,
+        dependencies: bitcoreModule.dependencies
+      });
```

Comment:
It might be nicer to just include the module that has dependencies and name fields on it. In the code we can iterate through the modules and get their name and dependencies. That way the config is easier.